src/NinjaTools.FluentMockServer.API/Configuration/ConfigFileProvider.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json.Linq;
+using NinjaTools.FluentMockServer.API.Models;
+using YamlDotNet.Serialization;
+
+namespace NinjaTools.FluentMockServer.API.Configuration
+{
+    /// <summary>
+    /// Runtime options for the MockServer
+    /// </summary>
+    public class MockServerOptions
+    {
+        public string ConfigFilePath { get; set; } = "/var/mock-server/config/";
+    }
+
+    public interface IConfigFileProvider
+    {
+        IEnumerable<IConfigFile> GetConfigFiles();
+    }
+
+   public class ConfigFileProvider : IConfigFileProvider
+    {
+        private readonly IFileSystem _fs;
+        private readonly ILogger<ConfigFileProvider> _logger;
+        private readonly MockServerOptions _options;
+
+        public ConfigFileProvider(IFileSystem fs, ILogger<ConfigFileProvider> logger, IOptions<MockServerOptions> options)
+        {
+            _fs = fs;
+            _logger = logger;
+            _options = options.Value;
+        }
+
+
+        public IEnumerable<IConfigFile> GetConfigFiles()
+        {
+            var rootDirectory = _options.ConfigFilePath;
+            var files = _fs.Directory.GetFiles(rootDirectory, "*", SearchOption.AllDirectories);
+
+            foreach (var file in files)
+            {
+                var fileExtension = _fs.Path.GetExtension(file).TrimStart('.').ToLower();
+
+                if (Enum.TryParse<ConfigurationFileType>(fileExtension, out var fileType))
+                {
+                    foreach (var configFile in ParseConfigFile(fileType, file))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'foreach' block.](https://app.codacy.com/manual/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4765163)